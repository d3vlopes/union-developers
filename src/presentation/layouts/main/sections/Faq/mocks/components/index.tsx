import { vitest } from 'vitest'

vitest.mock('@/presentation/components/organisms', () => ({
  QuestionsAccordion: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return (
      <div data-testid="mock-questions-accordion">{children}</div>
    )
  },
}))
