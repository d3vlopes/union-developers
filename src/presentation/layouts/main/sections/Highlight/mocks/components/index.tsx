import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Highlights: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-highlights">{children}</div>
  },
}))
