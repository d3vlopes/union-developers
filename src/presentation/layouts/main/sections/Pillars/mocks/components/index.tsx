import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  MethodologyCards: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-methodology-cards">{children}</div>
  },
}))
