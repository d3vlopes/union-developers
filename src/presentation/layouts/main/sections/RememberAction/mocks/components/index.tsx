import { vitest } from 'vitest'

vitest.mock(
  '@/presentation/components/molecules/RememberAction',
  () => ({
    RememberAction: function Mock({
      children,
    }: {
      children: React.ReactNode
    }) {
      return <div data-testid="mock-remember-action">{children}</div>
    },
  }),
)
