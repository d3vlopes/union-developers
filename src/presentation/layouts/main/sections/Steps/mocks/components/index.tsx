import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Steps: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="mock-steps">{children}</div>
  },
}))
