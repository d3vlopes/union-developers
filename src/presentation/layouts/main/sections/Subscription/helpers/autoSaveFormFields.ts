import { Dispatch, SetStateAction } from 'react'

export function autoSaveFormFields<TFormValues, TFormData>(
  formValues: TFormValues,
  formData: TFormData,
  errors: object,
  setFormData: Dispatch<SetStateAction<TFormData>>,
  delay = 5000,
) {
  const interval = setInterval(() => {
    const isFieldContainError = Object.keys(errors).length > 0

    if (!isFieldContainError) {
      setFormData({ ...formData, ...formValues })
    }
  }, delay)

  return interval
}
