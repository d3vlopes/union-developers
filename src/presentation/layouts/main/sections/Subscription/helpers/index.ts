export * from './autoSaveFormFields'
export * from './formStepMap'
