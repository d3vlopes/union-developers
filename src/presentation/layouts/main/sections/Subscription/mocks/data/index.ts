import { generateId } from '@/utils'

import { termsMock } from '../../steps/AboutYou/components/Terms/mock'

import { SubscriptionSectionProps } from '../..'

export const subscriptionMock: SubscriptionSectionProps = {
  id: 'formulario',
  title: 'Junte-se ao time!',
  initialStepDescription:
    'Tem interesse em participar do Union? Confira abaixo os requisitos para participar',
  formStepsDescription:
    'Ficamos felizes em saber que você tem interesse em fazer parte do Union! Agora é só responder o nosso formulário',
  requirements: [
    'Ter mais que 18 anos',
    'Ter 1 ano ou mais de estudo em programação',
    'Ainda não trabalhar na área',
    'Já ter criado projetos sozinho que não sejam de cursos ou eventos',
    'Não participar de bootcamps ou programas de capacitação',
    'Ter disponibilidade para trabalhar no mínimo 1 hora por dia no projeto',
    'Ter disponibilidade para participar das reuniões do time, durante a semana e finais de semana',
  ],
  buttonText: 'Inscrever-se',
  formStepNavLinks: [
    {
      id: generateId(),
      stepTitle: 'Dados pessoais',
      stepNumber: 1,
    },
    {
      id: generateId(),
      stepTitle: 'Informações técnicas',
      stepNumber: 2,
    },
    {
      id: generateId(),
      stepTitle: 'Sobre você',
      stepNumber: 3,
    },
    {
      id: generateId(),
      stepTitle: 'Enviar',
      stepNumber: 4,
    },
  ],
  terms: termsMock,
}
