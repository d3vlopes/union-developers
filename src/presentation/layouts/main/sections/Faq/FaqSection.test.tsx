import { render, screen } from '@/__tests__/helpers'

import { mainLayoutMock } from '@/presentation/layouts/main/mocks/data'

import './mocks'

import { FaqSection } from '.'

const { getByRole, getByTestId } = screen

const mock = mainLayoutMock.faqSection

describe('<FaqSection />', () => {
  beforeEach(() => {
    render(<FaqSection {...mock} />)
  })

  it('should contain correct section id', () => {
    const { container } = render(<FaqSection {...mock} />)

    const wrapper = container.firstChild

    expect(wrapper).toHaveAttribute('id', mock.id)
  })

  it('should render heading', () => {
    const heading = getByRole('heading', { name: mock.heading })

    expect(heading).toBeInTheDocument()
  })

  it('should render QuestionsAccordion', () => {
    const questionsAccordion = getByTestId('mock-questions-accordion')

    expect(questionsAccordion).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<FaqSection {...mock} />)

    expect(container).toMatchSnapshot()
  })
})
