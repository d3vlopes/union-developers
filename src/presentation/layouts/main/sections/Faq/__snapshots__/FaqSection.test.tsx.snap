// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FaqSection /> > should match snapshot 1`] = `
.c1 {
  position: relative;
  max-width: 125.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  font-size: 2rem;
  font-weight: 700;
  background: linear-gradient(225deg,#0048FE -50%,#851D86 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c2 {
  position: absolute;
  display: block;
  top: 50%;
  left: 0%;
  width: 30%;
  height: 3px;
  background: linear-gradient(225deg,#0048FE -70%,#851D86 100%);
  border-radius: 20px;
}

.c4 {
  position: absolute;
  display: block;
  top: 50%;
  right: 0%;
  width: 30%;
  height: 3px;
  background: linear-gradient(225deg,#0048FE 0%,#851D86 200%);
  border-radius: 20px;
}

.c0 {
  width: 100%;
  margin: 5.6rem 0;
  -webkit-scroll-margin-block-start: 150px;
  -moz-scroll-margin-block-start: 150px;
  -ms-scroll-margin-block-start: 150px;
  scroll-margin-block-start: 150px;
  -webkit-scroll-margin-block-end: 150px;
  -moz-scroll-margin-block-end: 150px;
  -ms-scroll-margin-block-end: 150px;
  scroll-margin-block-end: 150px;
}

.c5 {
  margin: 3.2rem auto 0;
  padding: 0;
}

@media (min-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

@media (min-width:320px) and (max-width:390px) {
  .c2 {
    width: 25%;
  }
}

@media (min-width:768px) {
  .c2 {
    width: calc(100% / 2 - 16.3rem);
    height: 5px;
  }
}

@media (min-width:320px) and (max-width:390px) {
  .c4 {
    width: 25%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: calc(100% / 2 - 16.3rem);
    height: 5px;
  }
}

@media (min-width:768px) {
  .c0 {
    margin-top: 15.5rem;
  }
}

@media (min-width:768px) {
  .c5 {
    margin: 5.6rem auto 0;
  }
}

<div>
  <section
    class="c0"
    id="questions"
  >
    <div
      class="c1"
    >
      <div
        aria-label="Linha a esquerda do titulo"
        class="c2"
      />
      <h2
        class="c3"
      >
        DÃºvidas
      </h2>
      <div
        aria-label="Linha a direita do titulo"
        class="c4"
      />
    </div>
    <div
      class="c5"
    >
      <div
        data-testid="mock-questions-accordion"
      />
    </div>
  </section>
</div>
`;
